test/handler/test_mypy_handler.vader

```diff
@@ -69,6 +71,67 @@ Execute(The mypy handler should parse lines correctly):
   \ '__init__.py:72:1: warning: Some warning',
   \ ])
 
+Execute(The mypy handler should parse `note` lines correctly, too):
+  call ale#test#SetFilename('__init__.py')
+
+  let g:ale_python_mypy_detect_notes = 1
+
+  AssertEqual
+  \ [
+  \   {
+  \     'lnum': 768,
+  \     'col': 38,
+  \     'filename': ale#path#Simplify(g:dir . '/baz.py'),
+  \     'type': 'E',
+  \     'text': 'Cannot determine type of ''SOME_SYMBOL''',
+  \   },
+  \   {
+  \     'lnum': 15,
+  \     'col': 3,
+  \     'filename': ale#path#Simplify(g:dir . '/__init__.py'),
+  \     'type': 'E',
+  \     'text': 'Argument 1 to "somefunc" has incompatible type "int"; expected "str"'
+  \   },
+  \   {
+  \     'lnum': 14,
+  \     'col': 0,
+  \     'filename': ale#path#Simplify(g:dir . '/another_module/bar.py'),
+  \     'type': 'I',
+  \     'text': 'In module imported here,',
+  \   },
```

Comment:
The notes provide additional information about previous problems, so we should instead include the lines in the `detail` key of the error above instead. Then using `:ALEDetail` or moving your mouse over an error will show you some more information about the error.